pkg/scalers/arangodb_scaler.go

```diff
@@ -128,48 +127,14 @@ func getNewArangoDBClient(meta *arangoDBMetadata) (driver.Client, error) {
 
 func parseArangoDBMetadata(config *scalersconfig.ScalerConfig) (*arangoDBMetadata, error) {
 	// setting default metadata
-	meta := arangoDBMetadata{}
-
-	// parse metaData from ScaledJob config
-	endpoints, err := GetFromAuthOrMeta(config, "endpoints")
-	if err != nil {
-		return nil, err
-	}
-	meta.endpoints = endpoints
-
-	if val, ok := config.TriggerMetadata["collection"]; ok {
-		meta.collection = val
-	} else {
-		return nil, fmt.Errorf("no collection given")
-	}
-
-	if val, ok := config.TriggerMetadata["query"]; ok {
-		meta.query = val
-	} else {
-		return nil, fmt.Errorf("no query given")
-	}
-
-	if val, ok := config.TriggerMetadata["queryValue"]; ok {
-		queryValue, err := strconv.ParseFloat(val, 64)
-		if err != nil {
-			return nil, fmt.Errorf("failed to convert queryValue to int, %w", err)
-		}
-		meta.queryValue = queryValue
-	} else {
-		if config.AsMetricSource {
-			meta.queryValue = 0
-		} else {
-			return nil, fmt.Errorf("no queryValue given")
-		}
+	meta := &arangoDBMetadata{}
+	meta.triggerIndex = config.TriggerIndex
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing arango metadata: %w", err)
 	}
 
-	meta.activationQueryValue = 0
-	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
-		activationQueryValue, err := strconv.ParseFloat(val, 64)
-		if err != nil {
-			return nil, fmt.Errorf("failed to convert activationQueryValue to int, %w", err)
-		}
-		meta.activationQueryValue = activationQueryValue
+	if !config.AsMetricSource && meta.QueryValue == 0 {
```

Comment:
Could config.AsMetricSource also be moved to meta?

Reply:
Hmm. All of the scaler config refactor PR has the same Changelog as mentioned in https://github.com/kedacore/keda/issues/5797#issuecomment-2117912592. Do you think it is better to add this `config.AsMetricSource` change to a separate PR with a dedicated changelog :thinking: ?

Reply:
Sure

Reply:
in general, I like https://github.com/kedacore/keda/pull/5819, but I would like to propose we wait for a few more configs refactored. It's possible there will be good number of scalers that find it useful and then we merge it, some configs will implement it as `_ scalersconfig.ScalerConfig` and all is good. But if we see that not many configs actually need it, then I think it makes sense we consider some alternative.